pkg/oauth/external/errors.go

```diff
@@ -0,0 +1,33 @@
+package external
+
+import (
+	"github.com/openshift/oauth-server/pkg/api"
+)
+
+type AuthorizationDeniedError struct {
+	Identity api.UserIdentityInfo
+	error
+}
+
+func NewAuthorizationDeniedError(identity api.UserIdentityInfo, err error) AuthorizationDeniedError {
+	return AuthorizationDeniedError{
+		Identity: identity,
+		error:    err,
+	}
+}
+
+func (e AuthorizationDeniedError) Unwrap() error { return e.error }
+
+type AuthorizationFailedError struct {
+	Identity api.UserIdentityInfo
```

Comment:
I'd rather this was exposed with a method

Reply:
done